libkbfs/md_journal.go

```diff
@@ -196,13 +196,15 @@ func makeMDJournalWithIDJournal(
 	}
 
 	if earliest != nil {
-		if earliest.BID() != latest.BID() {
+		if earliest.BID() != latest.BID() &&
+			!(earliest.BID() == NullBranchID &&
```

Comment:
Okay, I think I figured out why I was so confused. the MDID journal entry having `isLocalSquash = true` is mutually exclusive with the MD having branch ID `LocalSquashBranchID`, right? That is, the MDs in the `LocalSquashBranchID` all have `isLocalSquash = false`, will eventually be replaced with a single MD with `isLocalSquash = true`.

I get it now, but I can see myself getting confused again, esp. with statements like:

```
if entry.IsLocalSquash && isLocalSquash {
```

Possible better names for `LocalSquashBranchID`: `PendingSquashBranchID`, `PendingLocalSquashBranchID`, `WillSquashLocallyBranchID`. And the `isLocalSquash` variable could be `pendingLocalSquash`, etc. Or if none of those names work, perhaps a comment somewhere addressing the potential confusion.